src/time_series_interface.jl

```diff
@@ -500,6 +491,28 @@ function clear_time_series!(owner::TimeSeriesOwners)
     return
 end
 
+"""
+This function must be called when a component or attribute is removed from a system.
+"""
+function prepare_for_removal!(owner::TimeSeriesOwners)
+    clear_time_series!(owner)
+    set_shared_system_references!(owner, nothing)
```

Comment:
Why is this necessary?

Reply:
To make sure that a removed component doesn't have any references from the system.